tools/codegen/codespec/provider_spec_model.go

```diff
@@ -0,0 +1,111 @@
+package codespec
+
+type ElemType int
+
+const (
+	Bool ElemType = iota
+	Float64
+	Int64
+	List
+	Map
+	Number
+	Object
+	Set
+	String
+)
+
+type CodeSpecification struct {
```

Comment:
since the package is codespec, we could simplify this name to model (as well as the file name).

Reply:
updated